src/core/_dt.h

```diff
@@ -0,0 +1,95 @@
+//------------------------------------------------------------------------------
+// Copyright 2020 H2O.ai
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//------------------------------------------------------------------------------
+#ifndef dt__DT_h
+#define dt__DT_h
+#include <cstdint>
+#include <string>
+#include <vector>
+
+typedef struct _object PyObject;
+
+class Buffer;
+class Column;
+class DataTable;
+class Groupby;
+class RowIndex;
+class WritableBuffer;
+
+struct CString;
+enum class LType : uint8_t;
+enum class SType : uint8_t;
+
+using std::size_t;
+using colvec = std::vector<Column>;
+using strvec = std::vector<std::string>;
```

Comment:
I like `_dt.h` quite a lot, may be also get rid of the duplicated `strvec` definitions in the code and include `_dt.h` instead? Right now there are at least 6 identical `strvec` definitions.

Reply:
Absolutely, my thinking is that this header file can be used whenever we need to forward-declare any other class in a header. As such, "_dt.h" could eventually be included nearly everywhere. However, this comprehensive refactoring is not the purpose of this PR. It's just something that can be adjusted in the future as we go.